src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -111,7 +111,13 @@ public function registerClasses(Definition $prototype, string $namespace, string
                 $r = $this->container->getReflectionClass($class);
                 $attribute = null;
                 foreach ($r->getAttributes(When::class) as $attribute) {
-                    if ($this->env === $attribute->newInstance()->env) {
+                    $attributeEnv = $attribute->newInstance()->env;
+
+                    if (\is_string($attributeEnv)) {
+                        $attributeEnv = [$attributeEnv];
+                    }
```

Comment:
IIRC this should do the job too:
```suggestion
                    $attributeEnv = (array) $attribute->newInstance()->env;
```

Reply:
I'm divided on this one. Of course it's shorter, but it's indeed "harder" to read. Is there any added-value to this notation?

Reply:
it's more compact, I like it :)

Reply:
I don't have a strong opinion so let's write it this way! Fixed :+1: